src/info/field_spec_info.c

```diff
@@ -89,6 +144,66 @@ FieldSpecInfo FieldSpecInfo_Deserialize(const MRReply *reply) {
     MRReply *error = MRReply_MapElement(reply, IndexError_ObjectName);
     RedisModule_Assert(error);
     info.error = IndexError_Deserialize(error);
+    // attribute used to determine field type
+    info.stats = FieldStats_Deserialize(info.attribute, reply);
 
     return info;
 }
+
+// Assuming the spec is properly locked before calling this function.
+size_t IndexSpec_VectorIndexSize(IndexSpec *sp) {
+  VectorIndexStats stats = IndexSpec_GetVectorIndexesStats(sp);
+  return stats.memory;
+}
+
+VectorIndexStats IndexSpec_GetVectorIndexStats(IndexSpec *sp, const FieldSpec *fs){
+  VectorIndexStats stats = {0};
+  RedisModuleString *vecsim_name = IndexSpec_GetFormattedKey(sp, fs, INDEXFLD_T_VECTOR);
+  VecSimIndex *vecsim = openVectorIndex(sp, vecsim_name, DONT_CREATE_INDEX);
+  if (!vecsim) {
+    return stats;
+  }
```

Comment:
Can this happen?
Again, consider assertion over `if` when you expect something to be true.
We want to be more aggressive on this

Reply:
indexes are intialized lazely.
This is normal when the index is opened for read

Reply:
Sure? Sounds weird to me since we know that the index exists at this point and the name is correct for sure as well (as opposed to such invocations in parsing stage

Reply:
If no documents were added to it yet, it doesn't exist. Same for all index types

Reply:
Ah for no docs as well, got you 👍